web-common/src/layout/app-store.ts

```diff
@@ -14,54 +17,86 @@ export interface ActiveEntity {
  * App wide store to store metadata
  * Currently caches active entity from URL
  */
-export interface AppStore {
+interface AppStore {
   activeEntity: ActiveEntity;
   previousActiveEntity: ActiveEntity;
 }
 
-// We should rewrite ActiveEntity using appScreen dervied store
-const { update, subscribe } = writable({
+export const appScreen = derived(page, ($page) => {
+  let activeEntity;
+  switch ($page.route.id) {
+    case "/(application)":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Home,
+      };
+      break;
+    case "/(application)/source/[name]":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Source,
+      };
+      break;
+    case "/(application)/model/[name]":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Model,
+      };
+      break;
+    case "/(application)/dashboard/[name]":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Dashboard,
+      };
+      break;
+    case "/(application)/dashboard/[name]/edit":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.MetricsDefinition,
+      };
+      break;
+    case "/(application)/welcome":
+      activeEntity = {
+        name: $page?.params?.name,
+        type: MetricsEventScreenName.Splash,
+      };
+      break;
+    default:
+      // Return home as default
+      activeEntity = { name: "", type: MetricsEventScreenName.Home };
+  }
+
+  appStore.setActiveEntity(
+    activeEntity.name,
+    ScreenToEntityMap[activeEntity.type]
+  );
+  return activeEntity;
+});
+
+// TODO: Do we still need this?
```

Comment:
You can add the previous entity name to appScreen and get rid of this. Call `httpRequestQueue.inactiveByName` directly in the body of `appScreen`.

Reply:
I want to keep `appScreen` clean as it's being heavily utilized across the application for telemetry. Better it just contains the current active entity. Will leave this as is for now.